src/pages/boundary/api-docs/[[...page]].tsx

```diff
@@ -0,0 +1,110 @@
+import SidebarSidecarLayout from 'layouts/sidebar-sidecar'
+import { OpenApiPageContents } from 'components/open-api-page'
+/* Used server-side only */
+import { cachedGetProductData } from 'lib/get-product-data'
+import { isDeployPreview } from 'lib/env-checks'
+import fetchGithubFile from 'lib/fetch-github-file'
+import {
+	getPathsFromSchema,
+	getPropsForPage,
+	processSchemaString,
+	processSchemaFile,
+} from 'components/open-api-page/server'
+
+const productSlug = 'boundary'
+const targetFile = {
+	owner: 'hashicorp',
+	repo: 'boundary',
+	path: 'internal/gen/controller.swagger.json',
+	ref: 'stable-website',
+}
+// The path to read from when running local preview in the context of the boundary repository
+const targetLocalFile = '../../internal/gen/controller.swagger.json'
+
+function ApiDocsView({ layoutProps, apiPageProps }) {
+	return (
+		<SidebarSidecarLayout {...layoutProps} sidecarSlot={null}>
+			<OpenApiPageContents
+				info={apiPageProps.info}
+				operationCategory={apiPageProps.operationCategory}
+				renderOperationIntro={apiPageProps.renderOperationIntro}
+			/>
+		</SidebarSidecarLayout>
+	)
+}
+
+export async function getStaticPaths() {
+	let schema
+	if (isDeployPreview()) {
+		schema = await processSchemaFile(targetLocalFile)
+	} else {
+		const swaggerFile = await fetchGithubFile(targetFile)
+		schema = await processSchemaString(swaggerFile)
+	}
+	const paths = getPathsFromSchema(schema)
+	return { paths, fallback: false }
+}
+
+export async function getStaticProps({ params }) {
```

Comment:
`getStatic` functions here are meant to get this page working. I'm planning on refining these further and determining what abstractions to make in subsequent work.

For example, if / when we port over https://cloud.hashicorp.com/api-docs/packer, I think we'll be in a better spot to pick the right abstractions.